src/main/java/com/amazon/opendistroforelasticsearch/sql/metrics/MetricType.java

```diff
@@ -0,0 +1,55 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.metrics;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public enum MetricType {
```

Comment:
so far my impression is you combined the name of the metric and its type in a this enum, which is contradictory to its name. And one consequence of that choice is having the second type field in it, which in my opinion *is* the real type. Another bad side effect is when you instantiate metrics, you get the name of the metric from the type. So I suggest replacing the integer values (for `type` field) with actual enum values, in other words, getting rid of the type field, and making this enum less granular. Have a different utility class where you have constant names for metrics as well as the mapping between the name and the type (many to one is fine).

Reply:
Another bad side effect of adding that subtype (`type` field) : no one will understand what those numbers mean, and when one adds a new type with the existing approach s/he won't know which number to assign to it.

Reply:
Good suggestion, will do it.